### DISCUSSION TITLE
Switching Between fullscreen Apps refreshes page.

### DISCUSSION THREAD

**Discussion by rlawrence89** [vercel/next.js#38425] [Help] [answered]
I have started noticing a peculiar behavior. When i navigate to a different fullscreen program e.g microsoft word, then back to chrome my page refreshes. 

Why does this happen and how do i prevent that? Specifically want to prevent that for a multi step form that i have create. 

Is there some kind of screen focus behavior i am missing with nextjs/ react?

**Comment by icyJoseph** [comment, marked as answer]
> When i navigate to a different fullscreen program e.g microsoft word, then back to chrome my page refreshes.

Is this on Mac? What kind of fullscreen are we talking about? I don't think I've seen this happen before.

> Is there some kind of screen focus behavior i am missing with nextjs/ react?

Not that I know of... but libraries like react-query or SWR do perform some actions when the tab regains focus. Perhaps there's some unintentional side effect out of user code?

Could you record a little video of the issue? And what packages are you using?

### Resolution

`next-auth` has a feature to revalidate the user session on window focus. This was causing state to be lost in various components.

Reposting comment buried under for visibility:

> What do you think about this? [nextauthjs/next-auth#3405](https://github.com/nextauthjs/next-auth/issues/3405), could that be a lead to chase?
> 
> Reposting contents of that [issue](https://github.com/nextauthjs/next-auth/issues/3405) for visibility:
> 
> > With the removal of clientMaxAge and keepAlive in favor of refetchInterval, every time a tab is switched, the session gets revalidated, thus updating a session state in SessionProvider which then causes to rerender all components using useSession.
> > Is it possible to disable this "revalidation" upon window refocusing?
> > Thanks in advance.

**Reply by rlawrence89** [reply]
This behavior happens on MAC and windows.

Here is my package.json
```json
{
  "private": true,
  "scripts": {
    "dev": "next dev -p 3001",
    "build": "next build",
    "start": "next start -p 3001"
  },
  "dependencies": {
    "@apollo/client": "^3.6.9",
    "@fullcalendar/bootstrap": "^5.10.1",
    "@fullcalendar/common": "^5.11.0",
    "@fullcalendar/daygrid": "^5.10.1",
    "@fullcalendar/interaction": "^5.11.0",
    "@fullcalendar/list": "^5.10.1",
    "@fullcalendar/moment": "^5.11.0",
    "@fullcalendar/react": "^5.11.1",
    "@fullcalendar/scrollgrid": "^5.10.1",
    "@fullcalendar/timegrid": "^5.11.0",
    "@graphql-tools/schema": "^8.3.2",
    "@headlessui/react": "^1.6.5",
    "@heroicons/react": "^1.0.5",
    "@tailwindcss/forms": "^0.4.0",
    "axios": "^0.27.2",
    "babel-plugin-transform-require-ignore": "^0.1.1",
    "formik": "^2.2.9",
    "formik-persist-state": "^0.0.3",
    "graphql": "^16.3.0",
    "graphql-ws": "^5.6.2",
    "jose": "^4.8.3",
    "jsonwebtoken": "^8.5.1",
    "ldapjs": "^2.3.1",
    "next": "^12.2.0",
    "next-auth": "^4.2.1",
    "next-transpile-modules": "^9.0.0",
    "pino": "^7.9.2",
    "pino-pretty": "^7.5.4",
    "prop-types": "^15.8.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-dropzone": "^12.0.4",
    "react-hook-form": "^7.31.2",
    "react-icons": "^4.4.0",
    "react-idle-timer": "^4.6.4",
    "react-select": "^5.3.2",
    "react-table": "^7.7.0",
    "react-tailwind-table": "^1.1.1",
    "react-tiny-virtual-list-oss": "^2.3.1",
    "stylis": "^4.1.1",
    "subscriptions-transport-ws": "^0.11.0",
    "yup": "^0.32.11"
  },
  "devDependencies": {
    "@tailwindcss/forms": "^0.5.2",
    "@types/jsonwebtoken": "^8.5.8",
    "@types/ldapjs": "^2.2.2",
    "@types/node": "17.0.4",
    "@types/react": "17.0.38",
    "@types/react-dom": "^18.0.5",
    "@types/react-table": "^7.7.12",
    "autoprefixer": "^10.4.0",
    "msw": "^0.42.3",
    "postcss": "^8.4.5",
    "prettier": "^2.5.1",
    "prettier-plugin-tailwindcss": "^0.1.1",
    "tailwindcss": "^3.0.7",
    "typescript": "4.5.4"
  },
  "msw": {
    "workerDirectory": "public"
  }
}
```

Here is a short video of the behavior. 

- I enter data into the form and progress. 
- Switch to a new tab.
- Switch back to my form and it resets.

Formik has an option to persist form data which i used as a band aid for now.

**Reply by icyJoseph** [reply]
What do you think about this? https://github.com/nextauthjs/next-auth/issues/3405, could that be a lead to chase?

Reposting contents of that [issue](https://github.com/nextauthjs/next-auth/issues/3405) for visibility:

> With the removal of clientMaxAge and keepAlive in favor of refetchInterval, every time a tab is switched, the session gets revalidated, thus updating a session state in SessionProvider which then causes to rerender all components using useSession.
> 
> Is it possible to disable this "revalidation" upon window refocusing?
> Thanks in advance.

**Reply by rlawrence89** [reply]
This is actually my issue.  Thank you. All i needed to do was to pass the refetchOnWindowFocus prop to my session provider. Now no more resetting.
